src/platform/silabs/SiWx917/CHIPCryptoPAL.h

```diff
@@ -0,0 +1,1703 @@
+ */
+bool IsBufferContentEqualConstantTime(const void * a, const void * b, size_t n);
+
+template <typename Sig>
+class ECPKey
+{
+public:
+    virtual ~ECPKey() {}
+    virtual SupportedECPKeyTypes Type() const  = 0;
+    virtual size_t Length() const              = 0;
+    virtual bool IsUncompressed() const        = 0;
+    virtual operator const uint8_t *() const   = 0;
+    virtual operator uint8_t *()               = 0;
+    virtual const uint8_t * ConstBytes() const = 0;
+    virtual uint8_t * Bytes()                  = 0;
+
+    virtual bool Matches(const ECPKey<Sig> & other) const
+    {
+        return (this->Length() == other.Length()) &&
+            IsBufferContentEqualConstantTime(this->ConstBytes(), other.ConstBytes(), this->Length());
+    }
+
+    virtual CHIP_ERROR ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length, const Sig & signature) const = 0;
+    virtual CHIP_ERROR ECDSA_validate_hash_signature(const uint8_t * hash, const size_t hash_length,
+                                                     const Sig & signature) const                                              = 0;
+};
+
+/**
+ * @brief Helper class for holding sensitive data that should be erased from memory after use.
+ *
+ * The sensitive data buffer is a variable-length, fixed-capacity buffer class that securely erases
+ * the contents of a buffer when the buffer is destroyed.
+ */
+template <size_t kCapacity>
+class SensitiveDataBuffer
+{
+public:
+    ~SensitiveDataBuffer()
+    {
+        // Sanitize after use
+        ClearSecretData(mBytes);
+    }
+
+    SensitiveDataBuffer & operator=(const SensitiveDataBuffer & other)
+    {
+        // Guard self assignment
+        if (this == &other)
+            return *this;
+
+        ClearSecretData(mBytes);
+        SetLength(other.Length());
+        ::memcpy(Bytes(), other.ConstBytes(), other.Length());
+        return *this;
+    }
+
+    /**
+     * @brief Set current length of the buffer
+     * @return Error if new length is exceeds capacity of the buffer
+     */
+    CHIP_ERROR SetLength(size_t length)
+    {
+        VerifyOrReturnError(length <= kCapacity, CHIP_ERROR_INVALID_ARGUMENT);
+        mLength = length;
+        return CHIP_NO_ERROR;
+    }
+
+    /**
+     * @brief Returns current length of the buffer
+     */
+    size_t Length() const { return mLength; }
+
+    /**
+     * @brief Returns non-const pointer to start of the underlying buffer
+     */
+    uint8_t * Bytes() { return &mBytes[0]; }
+
+    /**
+     * @brief Returns const pointer to start of the underlying buffer
+     */
+    const uint8_t * ConstBytes() const { return &mBytes[0]; }
+
+    /**
+     * @brief Constructs span from the underlying buffer
+     */
+    ByteSpan Span() const { return ByteSpan(ConstBytes(), Length()); }
+
+    /**
+     * @brief Returns capacity of the buffer
+     */
+    static constexpr size_t Capacity() { return kCapacity; }
```

Comment:
This header duplicates src/crypto/CHIPCryptoPAL.h almost entirely. Generic changes like the new method Capacity() can be incorporated into the common code, and the rest should be added in a platform-specific header extending the common classes.

Reply:
Ohh okay, removed on the latest commit. Using the src/crypto/CHIPCryptoPAL.h directly

Reply:
@rcasallas-silabs modified based on the comments on the latest commit